File: medico/Services/Medico.Api/ClientApp/src/app/_models/selectableItemRequest.ts
import { SelectableItemType } from './selectableItemType';
import { SelectableVariableType } from './selectableVariableType';

export class SelectableItemRequest {
    type: SelectableItemType;
    selectableListId: string | null;
    minRangeValue: number | null;
    maxRangeValue: number | null;
    dateFormat: string | null;
    variableName: string | null;
    variableType: SelectableVariableType | null;
    variableInitialValue: string | null;

    toQueryParams(): any {
        const queryParams = {};
        const entityFilterFields = Object.keys(this);

        if (!entityFilterFields.length)
            return queryParams;

        for (let i = 0; i < entityFilterFields.length; i++) {
            const entityFilterField = entityFilterFields[i];
            queryParams[entityFilterField] = this[entityFilterField];
        }

        return queryParams;
    }
}

File: medico/Application/Medico.Application/SelectableItemsManagement/SelectableItemRequest.cs
using System;

namespace Medico.Application.SelectableItemsManagement
{
    public class SelectableItemRequest
    {
        public SelectableItemType Type { get; set; }

        public Guid? SelectableListId { get; set; }

        public double? MinRangeValue { get; set; }

        public double? MaxRangeValue { get; set; }

        public string DateFormat { get; set; }

        public string VariableName { get; set; }

        public SelectableVariableType? VariableType { get; set; }

        public string VariableInitialValue { get; set; }
    }
}